data/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { createStore } from 'redux';
+import { flowRight } from 'lodash';
+
+/**
+ * Module constants
+ */
+const reducers = [];
+const enhancers = [];
+if ( window.__REDUX_DEVTOOLS_EXTENSION__ ) {
+	enhancers.push( window.__REDUX_DEVTOOLS_EXTENSION__() );
+}
+const store = createStore( dynamicReducer, {}, flowRight( enhancers ) );
+
+/**
+ * Reducer function combining the dynamic "reducers" array
+ *
+ * @param {Object} state  Global state
+ * @param {Object} action Action
+ *
+ * @return {Object}       Updated global state
+ */
+function dynamicReducer( state = {}, action ) {
+	let hasChanges = false;
+	const newState = {};
+	reducers.forEach( ( { key, reducer } ) => {
+		const newSubState = reducer( state[ key ] || {}, action );
```

Comment:
Is the fallback of an object correct here? Wouldn't this mean the sub-reducer's own default value for state would never take effect? I would think we'd want to pass `undefined` if it's not yet in `state`.

Reply:
See also: https://github.com/reactjs/redux/blob/87071fd4ab71acc4fdd8b3db37d2d7ff08b724a3/src/combineReducers.js#L165-L180

(Not sure if you already pulled some from this implementation, since it's otherwise very similar)

Reply:
No, I didn't pull from this implement. Indeed, it's a good inspiration. I'll make some improvements once we settle on the bigger questions.